### PR TITLE
[prometheus-mysql-exporter] Allow overriding of image registry and add quay.io default

### PR DESCRIPTION
**Pull request by jnoordsij**
#### What this PR does / why we need it

Allow setting `image.registry` value to specify the registy the image should be pulled from.
Also makes the default to be `quay.io`, which seems to be more consistent with various other charts in this repository.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-mysql-exporter/values.yaml [hunk 1]
```diff
@@ -8,7 +8,8 @@ fullnameOverride: ""
 replicaCount: 1
 
 image:
-  repository: "prom/mysqld-exporter"
+  registry: quay.io
+  repository: prometheus/mysqld-exporter
```
Comment (monotek) [inline charts/prometheus-mysql-exporter/values.yaml:12]: when we change default of an existing variable we should raise major version and add some info in the upgrading section of the chart.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-mysql-exporter/values.yaml (hunk 1 above)
Comment (jnoordsij) [reply charts/prometheus-mysql-exporter/values.yaml:12]: I modelled partially after #928 which did basically the same for the `alertmanager` chart in a minor bump. The adding of the `registry` key does make this slightly different though. You really think a major version bump is the way to go here? Then I'll add some notes to the `README`.

**Review by monotek — APPROVED** [review]

**Comment by Jeroen0494** [comment]
This PR breaks my current installation:
```
  Warning  Failed     5m54s (x4 over 7m37s)   kubelet            Failed to pull image "quay.io/prom/mysqld-exporter:v0.14.0": rpc error: code = Unknown desc = failed to pull and unpack image "quay.io/prom/mysqld-exporter:v0.14.0": failed to resolve reference "quay.io/prom/mysqld-exporter:v0.14.0": pulling from host quay.io failed with status code [manifests v0.14.0]: 401 UNAUTHORIZED
```

Are you sure the path to the image is correct?

**Comment by Jeroen0494** [comment]
Seems this PR required fixes on the user side. My repository was by default set to `prom/mysqld-exporter` instead of `prometheus/mysqld-exporter`.

**Pull request merged**